src/device/gcn/array.jl

```diff
@@ -133,3 +133,20 @@ end
         nothing
     end
 end
+
+## reshape
+
+function Base.reshape(a::ROCDeviceArray{T,M,A}, dims::NTuple{N,Int}) where {T,N,M,A}
+#   if prod(dims) != length(a)
```

Comment:
Proper exception throwing requires fixes to work on Julia 1.11+.
For now I think we can do `throw(nothing)` to at least signal an error that silently skipping it.